### ISSUE TITLE
Make Font Awesome Icons accessible to visually impaired campers

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#15739]
A lot of campers who are working through freeCodeCamp are visually impaired or completely blind. We should use additional HTML elements everywhere where we use a Font Awesome icon that explains what the icon means.

Font Awesome has a guide for this here: http://fontawesome.io/accessibility/

**Comment by anawebdev** [comment]
Hi! Can i give it a try?

**Comment by sblasa** [comment]
@QuincyLarson @anawebdev Can I also help out on this issue? I would happy to assist. It would be my first time.

**Comment by anawebdev** [comment]
@sblasa Sure, that would be great!

Can anyone confirm if we can start working on this?

**Comment by raisedadead** [comment]
@sblasa @anawebdev yes sure go for it, help wanted issues are open for all to explore and fix.! Hit us up in the [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors), if you need any assistance.

**Comment by xnt** [comment]
I have a bit of experience doing aria-* stuff on my 9to5. @sblasa @anawebdev are you already on it?

**Comment by sblasa** [comment]
Hi! @xnt I'm working on this with @anawebdev! Thanks!

**Comment by QuincyLarson** [comment]
@xnt We'd welcome your help with some other help-wanted issues: https://github.com/FreeCodeCamp/freecodecamp/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22

**Comment by xnt** [comment]
@QuincyLarson yup! Already working on #13013 . Almost done (lint'ing). Thanks!

**Comment by QuincyLarson** [comment]
@sblasa @anawebdev Have you had time to make any progress on this?

**Comment by sblasa** [comment]
@QuincyLarson @raisedadead @anawebdev Hi, sorry for the delay. I have my local freeCodeCamp working now and starting on it. I read the directions for the Font Awesome guide and understand what to do.  Is there a particular area/section or directory I need to start in freeCodeCamp?  Thanks!

**Comment by raisedadead** [comment]
Hi @sblasa!

Thanks for your interest in working on this.

Most pugjs (.jade) files should be in 
https://github.com/freeCodeCamp/freeCodeCamp/tree/staging/server/views

Of course, some client side views are now using react, so you may have to look through some `.jsx` files in https://github.com/freeCodeCamp/freeCodeCamp/tree/staging/common/app, but the views directory should get you started.

Also, there is this old issue (https://github.com/freeCodeCamp/freeCodeCamp/issues/13798) that was closed recently but it hasn't been implemented. Feel free to work on that since you are at it. The thread has more relevant info.

As always feel free to reach out to us here or on the chat.

**Comment by sblasa** [comment]
Hi @raisedadead, okay, thank you for letting me know. I'm going to look at issue #13798 in more detail, and I saw the list that was posted there, so I will refer to that.

**Comment by QuincyLarson** [comment]
@sblasa We would welcome your help on this issue and on #13798 (which should probably be done first).

**Comment by sblasa** [comment]
@QuincyLarson @raisedadead  okay, so I should work with @BhaveshSGupta on this #13798  and do that issue first with the list that @BhaveshSGuptaI created. I just glanced through the issue just now. Thanks!

**Comment by QuincyLarson** [comment]
@sblasa Yes - you can reach out to @BhaveshSGupta (Gitter may be a good place to do this) and see if you can help him. Either way, you should probably wait until he's finished #13798 before attempting to start this.

**Comment by sblasa** [comment]
@bhaveshsgupta hi, are you still working on issue #13798? Once you're done, I will fix issue #15739 Make Font Awesome Icons accessible to visually impaired campers.  Let me know if you'd like me to help 😀 Thanks!

**Comment by sblasa** [comment]
@QuincyLarson thanks, I've reached out to him via Gitter too. In any case, I will wait for him to fix the issue #13798.  😀

**Comment by adikash1997** [comment]
Hii, Can I help in this issue?

**Comment by sblasa** [comment]
@QuincyLarson @bhaveshsgupta was the other issue #13798 fixed? I was just wondering when I could start work with this along with  @adikash1997 help.

Thanks!

**Comment by QuincyLarson** [comment]
@sblasa We are still in the process of doing this. Thanks for your patience.

**Comment by QuincyLarson** [comment]
@anawebdev @adikash1997 @sblasa We just launched our updated coding platform. Would any of you still be interested in tackling this accessibility issue?

**Comment by sblasa** [comment]
Hi @QuincyLarson ! I'm still interested--just need to review the directions again. :smile:

**Comment by QuincyLarson** [comment]
Hi @sblasa did you get a chance to make any progress on this? Let us know if you have any questions or get stuck on anything :)

**Comment by xnt** [comment]
I'm working on this after chatting with @QuincyLarson . I think I'm pretty much there. 

However, I'm blocked from actually testing most changes because of #17382 - workaround suggestions are appreciated!

**Comment by sblasa** [comment]
@QuincyLarson , @xnt Okay, thanks for letting me know. Let me know if you need assistance with anything else. :thumbsup:

**Comment by xnt** [comment]
Please note that testing this is blocked by #17382 , and I don't want to create a PR without testing. I'm not versed enough on FCC as to come up with a workaround by my own, either. So right now I've pushed stuff to https://github.com/xnt/freeCodeCamp/tree/fontawesome-a11y waiting to be able to test.

**Comment by QuincyLarson** [comment]
@sblasa Once we've fixed  #17382  we would welcome your help with this!

**Comment by OMahonyD** [comment]
Hi @QuincyLarson, I've recently been reactivated in the coding world & have some experience with aria roles etc. Let me know if I can help with #a11y (accessibility) issues. Dan

**Comment by xnt** [comment]
@OMahonyD I wanna look into this when I have a chance, but if you feel compelled to it, feel free to merge the branch mentioned above. Should at the very least be a good starting point.

**Comment by QuincyLarson** [comment]
@OMahonyD Thanks for your kind offer. Yes - we would welcome your help with this! Would you be interested in tackling this issue? If so, we can assign it to you.

**Comment by thecodingaviator** [comment]
@xnt Are you interested in going ahead right now?

**Comment by xnt** [comment]
@thecodingaviator W00t!! The blocking issue is fixed now! 

I'll check if my branch still merges smoothly, probably tomorrow. 

Thanks for the heads up.
